### PR TITLE
[Feature:Developer] Save VM after vagrant up job

### PR DESCRIPTION
**Pull request by IDzyre**
### Please check if the PR fulfills these requirements:

* [ ] Tests for the changes have been added/updated (if possible)
* [ ] Documentation has been updated/added if relevant
* [ ] Screenshots are attached to Github PR if visual/UI changes were made

### What is the current behavior?
Submitty has a daily vagrant up test that ensures that developers can successfully run vagrant up. 

### What is the new behavior?
That vagrant box will now be 'packaged' , and uploaded to "vagrant cloud"
This box is for vagrant and virtualbox, if using any other software (QEMU, libvirt, etc) it will use the base ubuntu box for that software, and they won't see the benefits of this (yet). 

### Other information?
related to #9723
related to #8906

### CODE REVIEW

File: .github/workflows/vagrant_up.yaml [hunk 1]
```diff
@@ -13,14 +13,16 @@ jobs:
...
     steps:
-      - uses: actions/checkout@v4
-      # Installing this is skipped in CI as using it was causing some corruption in
-      # attempting to mount the shared folder. The guest and host guest additions
-      # versions are close enough that we are fine without it.
-      # - run: vagrant plugin install vagrant-vbguest
-      - run: NO_SUBMISSIONS=1 vagrant up ubuntu-22.04
-      - name: Validate image
-        run: curl --show-error --fail --include http://localhost:1511
+      - uses: actions/checkout@v3
```
Comment (williamjallen) [inline .github/workflows/vagrant_up.yaml:16]: It looks like you might have had a bad merge conflict here.  This should be bumped to `v4` to match the latest version on main.

**Review by williamjallen — CHANGES_REQUESTED** [review]

File: .github/workflows/vagrant_up.yaml [hunk 2]
```diff
@@ -11,16 +11,35 @@ permissions:
...
+          fi
+          echo "version=${TAG:1}" >> $GITHUB_OUTPUT
+
       # Installing this is skipped in CI as using it was causing some corruption in
       # attempting to mount the shared folder. The guest and host guest additions
       # versions are close enough that we are fine without it.
       # - run: vagrant plugin install vagrant-vbguest
-      - run: NO_SUBMISSIONS=1 vagrant up ubuntu-22.04
+      - name: Vagrant Up
+        run: NO_SUBMISSIONS=1 vagrant up
```
Comment (williamjallen) [inline .github/workflows/vagrant_up.yaml:33]: Is there enough disk space to make submissions?  Most developers who are downloading this VM are going to want the sample courses to be set up already.

**Review by ziesski — APPROVED** [review]
I tested this feature with my intel Mac, with virtual box.
This was very smooth and fast!
With my old intel mac, vagrant destroy and up takes at least 1.5 hours, but with this PR it took less than 10 mins on initial up, and from then on it takes less than 5 mins for VM to be up and running. (as it skips the download process). 
One thing to mention tho is that it ends without any notification/prompt(green duck). 
So I was slightly confused when it terminated. Should add some info when process completes.

**Review by VVVITH — APPROVED** [review]
Test with Arm M1 chip Mac using qemu, it goes well now.

**Pull request merged**